### PR TITLE
Signed Settlements

### LINKED ISSUE #34: Investigate missing polyfill error in github actions
Investigate why tests pass locally but this one test fails inside the GithubCI with the missing polyfill error, because `reflect-metadata` hasn't been imported yet even though it gets imported.

### LINKED ISSUE #142: Add support for signed settlements / bridging / dispatching
**Is your feature request related to a problem? Please describe.**
When running the Protokit sequencer with proofs disabled, the standard settlement contract would not accept the bundle proof (merged blocks into a single proof) due to not being authorized by a real proof. In order to integrate with lightnet or any other live networks, we need signed settlements.

**Describe the solution you'd like**
The permissions on all affected contracts should change to "signature" instead of "proof" for this specific type of operation. This results to being able to deploy the same contract with various permissions depending on the use case (proofs on or off)

**Describe alternatives you've considered**
N/A

**Additional context**
This task is a prerequisite for the lightnet integration into the starter kit.

### LINKED ISSUE #144: Add lightnet support into sequencer tests for settlement/bridging/dispatching
**Is your feature request related to a problem? Please describe.**
As of now all sequencer tests that touched the L1 in any way were ran using LocalBlockchain. In order to proceed with a more production-like integrations, we need to ensure all the sequencer code (including components like BaseLayer) work correctly with a production-like network.

**Describe the solution you'd like**
Integrate lightnet into the test suite of the sequencer and ensure all relevant components work as they previously did with LocalBlockchain. Components may include: BaseLayer, settlement contract, dispatcher contract

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by rpanic**
Closes #142 
Closes #144 
Closes #34 

This PR implements signed settlements for the purpose of testing settlement and bridging against lightnet or devnet.
It automatically discovers the mode through the combination of the type of network and whether proofs are enabled.
Based on that, it sets the permissions of the deployed contracts and sets the authorization type of the accountupdates to these contracts. 

Things introduced in this PR:
Permission sets for settlement contracts through `BaseLayerContractPermissions`
`FeeStrategy` interface, which modularizes what number the sequencer uses for L1 fees

Changed:
The settlement contract tracks the block where the last settlement occurred internally. Previously, this was the `globalSlotSinceGenesis`, but this lead to issues. Mainly the retrieval of events is based on the blockheight, and any mapping from slot => height would be non-trivial. Therefore this internal tracker was changed to the `blockheight` (or `chainLength` L1 network precondition) value. 
Possible attack vectors are potentially by L1 validators to skip production of blocks in won slots in order to lengthen the escape hatch window. But this is a reasonable assumption to make about the settlement layer and should lead to very short delays at worst, which doesnt affect any L2 properties

### CODE REVIEW

**Review by ejMina226 — APPROVED** [review]

File: .github/workflows/pull-request-develop.yml [hunk 1]
```diff
@@ -115,6 +115,19 @@ jobs:
...
+        env:
+          RUN_ARCHIVE_NODE: true
+          LOG_LEVEL: 'INFO'
+          PROOF_LEVEL: 'none'
+          NETWORK_TYPE: 'single-node'
+        ports:
+          - 3085:3085
+          - 8080:8080
+          - 8181:8181
+          #     archive endpoints
```
Comment (omergulen) [inline .github/workflows/pull-request-develop.yml:129]:
```suggestion
          # archive endpoints
```

File: .github/workflows/pull-request-develop.yml [hunk 2]
```diff
@@ -132,5 +145,12 @@ jobs:
...
 
+      - name: "Wait for lightnet readiness"
+        uses: o1-labs/wait-for-mina-network-action@v1
+        with:
+          mina-graphql-port: 3085
+          max-attempts: 30
+          polling-interval-ms: 5000
+
       - name: "Integration tests"
         run: npm run test:integration
```
Comment (omergulen) [inline .github/workflows/pull-request-develop.yml:156]:
```suggestion
        run: npm run test:integration

```

File: packages/deployment/docker/docker-compose.yml [hunk 3]
```diff
@@ -7,3 +7,5 @@ include:
   - ./sequencer/docker-compose.yml
   # Enabled by worker profile
   - ./worker/docker-compose.yml
+
+  - ./lightnet/docker-compose.yml
```
Comment (omergulen) [inline packages/deployment/docker/docker-compose.yml:11]:
```suggestion
  - ./lightnet/docker-compose.yml

```

File: packages/deployment/docker/lightnet/docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+      - LOG_LEVEL=INFO
+      - PROOF_LEVEL=none
+      - NETWORK_TYPE=single-node
+    ports:
+      - 3085:3085
+      - 8080:8080
+      - 8181:8181
+#     archive endpoints
+      - 5432:5432
+      - 8282:8282
```
Comment (omergulen) [inline packages/deployment/docker/lightnet/docker-compose.yml:16]:
```suggestion
      - 8282:8282

```

File: .github/workflows/pull-request-develop.yml (hunk 1 above)
Comment (omergulen) [reply .github/workflows/pull-request-develop.yml:129]: **nitpick**

File: .github/workflows/pull-request-develop.yml (hunk 2 above)
Comment (omergulen) [reply .github/workflows/pull-request-develop.yml:156]: **nitpick**

File: packages/deployment/docker/docker-compose.yml (hunk 3 above)
Comment (omergulen) [reply packages/deployment/docker/docker-compose.yml:11]: **nitpick**

File: packages/deployment/docker/lightnet/docker-compose.yml (hunk 4 above)
Comment (omergulen) [reply packages/deployment/docker/lightnet/docker-compose.yml:16]: **nitpick**

**Review by maht0rz — APPROVED** [review]

**Pull request merged**